src/lib/colorUtils.ts

```diff
@@ -0,0 +1,50 @@
+import {ColorRgb, Option} from "@/types/common.types";
+
+const RED = 0.2126;
+const GREEN = 0.7152
+const BLUE = 0.0722;
+const GAMMA = 2.4;
+
+export function hexToRgb(hexValue: string): Option<[number, number, number]> {
+  const result = /^#?([a-fA-F\d]{2})([a-fA-F\d]{2})([a-fA-F\d]{2})$/.exec(hexValue);
+  return result ? [
+    parseInt(result[1],16),
+    parseInt(result[2], 16),
+    parseInt(result[3], 16)
+  ] : null;
+}
+
+export function rgbToHex(color: ColorRgb): string {
+  const [r,g,b] = color;
+  return "#" + r.toString(16) + g.toString(16) + b.toString(16);
+}
+
+export function getLuminance(color: ColorRgb) {
+  const a = [...color].map(v => {
+    v /= 255;
+    return v <= 0.03928 ? v / 12.92 : Math.pow((v + 0.055) / 1.055, GAMMA);
```

Comment:
what do these magic numbers  mean, 
put these in consts with a good name and/or an explanation